packages/governance/src/typeGuards.js

```diff
@@ -0,0 +1,285 @@
+export const SimplePositionsShape = [
+  YesSimplePositionShape,
+  NoSimplePositionShape,
+];
+export const SimpleIssueShape = SimpleSpecShape;
+export const SimpleQuestionSpecShape = harden({
+  method: ChoiceMethodShape,
+  issue: SimpleIssueShape,
+  positions: SimplePositionsShape,
+  electionType: M.or('election', 'survey'),
+  maxChoices: M.gte(1),
+  closingRule: ClosingRuleShape,
+  quorumRule: QuorumRuleShape,
+  tieOutcome: NoSimplePositionShape,
+});
+export const SimpleQuestionDetailsShape = harden({
+  method: ChoiceMethodShape,
+  issue: SimpleIssueShape,
+  positions: SimplePositionsShape,
+  electionType: M.or('election', 'survey'),
+  maxChoices: M.gte(1),
+  closingRule: ClosingRuleShape,
+  quorumRule: QuorumRuleShape,
+  tieOutcome: NoSimplePositionShape,
+  questionHandle: makeHandleShape('Question'),
+  counterInstance: InstanceShape,
+});
+
+export const SimplePositionsShapeA = [
+  harden({ text: 'yes' }),
+  harden({ text: 'no' }),
+];
+
+export const SimpleQuestionSpecShapeA = harden({
+  positions: SimplePositionsShape,
+  tieOutcome: NoSimplePositionShape,
+});
+
+export const QuestionSpecShape = M.or(
+  ApiInvocationQuestionSpecShape,
+  OfferFilterQuestionSpecShape,
+  ParamChangesQuestionSpecShape,
+  SimpleQuestionSpecShape,
+);
+
+export const PositionShape = M.or(
+  YesApiInvocationPositionShape,
+  NoApiInvocationPositionShape,
+  YesOfferFilterPositionShape,
+  NoOfferFilterPositionShape,
+  YesSimplePositionShape,
+  NoSimplePositionShape,
+  YesParamChangesPositionShape,
+  NoParamChangesPositionShape,
+);
+
+export const QuestionHandleShape = makeHandleShape('question');
+
+// TODO(hibbert): add details; move to a more appropriate location
+export const SubscriberShape = M.remotable('Subscriber');
+export const InvitationShape = M.remotable('Invitation');
+
+// XXX I want to add questionHandle and counterInstance to
+// ParamChangesQuestionSpecShape. I don't see any alternative to adding the
+// methods to each member separately
+export const QuestionDetailsShape = M.or(
+  ParamChangesQuestionDetailsShape,
+  ApiInvocationQuestionDetailsShape,
+  OfferFilterQuestionDetailsShape,
+  SimpleQuestionDetailsShape,
+);
+
+export const ElectoratePublicI = M.interface('Committee PublicFacet', {
+  getQuestionSubscriber: M.call().returns(SubscriberShape),
+  getOpenQuestions: M.call().returns(M.promise()),
+  getName: M.call().returns(M.string()),
+  getInstance: M.call().returns(InstanceShape),
+  getQuestion: M.call(QuestionHandleShape).returns(M.promise()),
+});
+
+export const ElectorateCreatorI = M.interface('Committee AdminFacet', {
+  getPoserInvitation: M.call().returns(M.promise()),
+  addQuestion: M.call(InstanceShape, QuestionSpecShape).returns(M.promise()),
+  getVoterInvitations: M.call().returns(M.arrayOf(M.promise())),
+  getQuestionSubscriber: M.call().returns(SubscriberShape),
+  getPublicFacet: M.call().returns(ElectoratePublicI),
```

Comment:
This hazard is on me, and I may eventually fix it. But at least for now an InterfaceGuard such as `ElectoratePublicI` is not itself a pattern.

```suggestion
const ElectoratePublicShape = M.remotable('ElectoratePublic');
...
  getPublicFacet: M.call().returns(ElectoratePublicShape),
```

Reply:
Done